client/state/user-settings/reducer.js

```diff
@@ -57,8 +57,22 @@ export const updatingPassword = ( state = false, action ) => {
 	return state;
 };
 
+export const updating = ( state = false, action ) => {
+	switch ( action.type ) {
+		case USER_SETTINGS_SAVE: {
+			return true;
+		}
+		case USER_SETTINGS_UPDATE:
+		case USER_SETTINGS_UPDATE_FAILURE: {
+			return false;
+		}
+	}
+	return state;
+};
+
 export default combineReducers( {
 	settings,
 	unsavedSettings,
 	updatingPassword,
+	updating,
```

Comment:
A side note that `updating`, `updatingPassword` and `unsavedSettings` feels like component state territory to me.  If we're handling those at a single location only, why do we need to pass them through Redux at all?

Reply:
I'm not sure I can follow. How would we indicate that a request is `pending` if that request is handled by the data layer? That seems to be a general pattern we're using throughout Calypso. I may be misunderstanding though.

Reply:
Yeah, I agree this would create additional work. Let's leave it as-is.